app/src/main/java/org/cuberite/android/MainActivity.java

```diff
@@ -326,6 +326,8 @@ public void onClick(DialogInterface dialog, int id) {
             builder.create().show();
         } else if(preferences.getString("cuberiteLocation", "").startsWith(PRIVATE_DIR)){
             Log.d(Tags.MAIN_ACTIVITY, "Cuberite has permissions for external storage, but is still in the private dir");
+        } else if(preferences.getString("cuberiteLocation", null) == null) {
+            preferences.edit().putString("cuberiteLocation", "/sdcard/cuberite-server").apply();
```

Comment:
I don't think a `checkPermission` function is a good place to be overriding any preferences.

Reply:
yeah, that is not a nice decision.
But the only place where i found, that preferences are set, is at this function: 
```onRequestPermissionsResult```
but this function is only indirectly called by ```checkPermission``` so the function is changing the preferences already.

Reply:
but maybe its useful to change the hardcoded path to something like this
```suggestion
preferences.edit().putString("cuberiteLocation", PUBLIC_DIR + "/cuberite-server").apply();
```